spec/features/navigation/navigation_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe 'Site Navigation' do
+  before :each do
+
+    visit '/'
+
+		click_link 'Register'
+
+		expect(current_path).to eq("/register")
+
+		name = "Happy Gilmore"
+		street_address = "45433 Fake st."
+		city = "Denver"
+		state = "Colorado"
+		zip = 80234
+		email = "strangerthings@gmail.com"
+		password = "BestFakerEver"
+		password_confirmation = "BestFakerEver"
+
+    fill_in :name, with: name
+    fill_in :street_address, with: street_address
+    fill_in :city, with: city
+    fill_in :state, with: state
+    fill_in :zip, with: zip
+    fill_in :email, with: email
+    fill_in :password, with: password
+    fill_in :password_confirmation, with: password_confirmation
+
+    click_on 'Create User'
+  end
+
+  it 'shows this information from the nav bar' do
+      within('#top-nav') do
+        expect(page).to have_link("Home")
+        expect(page).to have_link("All Merchants")
+        expect(page).to have_link("All Items")
+        expect(page).to have_link("Cart")
+        expect(page).to have_link("Login")
+        expect(page).to have_link("Register as New User")
+        expect(page).to have_content("Cart: 0")
```

Comment:
They already have a instance method in the cart model for counting total count of the cart